File: Assignment_1/Q2.cpp
#include <iostream>            //  for cout
#include <cmath>              // for NAN

using namespace std;

double factorial(int);                    // Declaration of the function factorial.

int main()
{

    cout << " Factorail of 10 is : ";              // Some sample testing of function factorial.
    cout << factorial(10) << endl;
    cout << " Factorail of 15 is : ";
    cout << factorial(15) << endl;
    cout << " Factorail of 5 is  : ";
    cout << factorial(5) << endl;
    cout << " Factorail of -5 is : ";
    cout << factorial(-5) << endl;

}
// -------------*  factorial function *-------------------------------
double factorial(int num)
{
    if ( num < 0.0 )            // This will check whether a number is negative,if it is nagative then it  will return nan.
    return NAN;


    double count = 1;
    for(int i=1;i<= num;i++)     //This is simple for loop for factorial
    {
        count *= i; 
    
    }
    return count;   
}

File: Assignment_1/python/Q2.py


#----------------------* main function *----------------------------------  
def main():
    print( " Factorail of 10 is : " )           #Some sample testing of function Factorial. 
    print( Factorial(10) )
    print( " Factorail of 15 is : " )
    print( Factorial(15) )
    print( " Factorail of 5 is  : " )
    print( Factorial(5) )
    print( " Factorail of -5 is : " )
    print( Factorial(-5) )



#------------------------* Factorial function *----------------------------
def Factorial(num):
    if (num < 0):                            # It will check whether number is negative or not.If it is the negative then it will return nan.
        return float("nan")
    coutn =1
    for i in range(1,num+1):
        coutn*=i
    return coutn

#-------------------------* If this program is running as a __main__ then it execute the main function *----------------
if __name__=="__main__":
    main()